backend/src/main/java/org/eclipse/tractusx/discoveryfinder/service/DiscoveryFinderService.java

```diff
@@ -25,10 +25,12 @@
 import org.eclipse.tractusx.discoveryfinder.model.Endpoint;
 import org.eclipse.tractusx.discoveryfinder.repository.EndpointRepository;
 import org.eclipse.tractusx.discoveryfinder.service.utils.UuidUtils;
+import org.springframework.context.annotation.Primary;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
 @Service
+@Primary
```

Comment:
Could you explain why the upstream service is the one to choose, on my first sight, this would mean, that it is selected always, right? Or is it not part of the distribution with your downstream additions?

Reply:
The `@Primary` annotation here is due to the way Spring selects components to instantiate. 
Adding the annotation on the upstream code means we don't need to suggest this installation anywhere else in the upstream code: every place where the Service class is `@Autowired` will keep working.

As we have two services with the same purpose (`DiscoveryFinderService` and `DownstreamDiscoveryFinderService`), it is more straightforward if we explicitly mention when we want to instantiate the downstream one. One example of that is on the `DownstreamFinderApiDelegate` file: `(@Autowired @Qualifier("downstreamService")`.